### ISSUE TITLE
Option to Setup initial state (G) as a function

### ISSUE THREAD

**Issue by FateRiddle** [google/boardgame.io#40]
Feature request that `G` takes into account meta data like `numPlayer`.
 
Currently it seems no way to initialize G with `ctx` as argument.

How about allowing G accept function that returns an object.
```js
// so instead of
G: {  numberOfPieces:  20 }
// we can do
G: (ctx) => ({ numberOfPieces: ctx.numPlayer * 10 })
```

**Comment by nicolodavis** [comment]
Great idea. Will add it to the list.

**Comment by FateRiddle** [comment]
Great! One more question that might not qualify as 'issue':

I notes both server and client requires "game rules" file (TicTacToe in example). Typically it is written in ES6, but I think node.js doesn't support es6 import yet? How do I make it work?

**Comment by nicolodavis** [comment]
Have a look at the package.json file. It uses babel-node as a development server. Alternatively, you could use a module bundler to convert down from ES6 when deploying.

**Comment by wishingandwandering1** [comment]
Having problems someone changing my name to another to collect gifts prizes and deposits

**Comment by nicolodavis** [comment]
@FateRiddle this is now implemented in 0.14.0
